src/main/scala/cognite/spark/v1/TimeSeriesRelation.scala

```diff
@@ -133,6 +137,19 @@ object TimeSeriesRelation extends UpsertSchema {
   val readSchema = structType[TimeSeriesReadSchema]
 }
 
+object LegacyNameSource extends Enumeration {
+  type LegacyNameSource = Value
+  val None, Name, ExternalId = Value
```

Comment:
We used uppercase Enumerations for the `OnConflict` handling, but I see Spark used Pascal for their `SaveMode` Enumerations. Perhaps you could change the existing ones while you're at it for consistency.